cmd/libsnap-confine-private/snap.c

```diff
@@ -31,7 +31,7 @@
 bool verify_security_tag(const char *security_tag, const char *snap_name)
 {
 	const char *whitelist_re =
-	    "^snap\\.([a-z0-9](-?[a-z0-9])*(_[a-z0-9]{1,10})?)\\.([a-zA-Z0-9](-?[a-zA-Z0-9])*|hook\\.[a-z](-?[a-z])*)$";
+	    "^snap\\.([a-z0-9](-?[a-z0-9])*(_[a-z0-9]{1,10})?)\\.([a-zA-Z0-9](-?[a-zA-Z0-9])*|hook\\.[a-z](-?[a-z0-9])*)$";
```

Comment:
Looks good and matches the characters we allow in `hooktypes.go`; nb, we could probably tweak the pattern to be even more consistent and say "....(-?[a-z0-9])+" instead of "....(-?[a-z0-9])*" (note "+" vs "*") cause we will never get a one-letter hook name here, but that doesn't improve/affect anything, so fine to leave as is.

Reply:
I'm happy to do it in another pass so that this is not a part of the stable release.